src/antsibull_changelog/copier.py

```diff
@@ -0,0 +1,129 @@
+# Author: Felix Fontein <felix@fontein.de>
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or
+# https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+# SPDX-FileCopyrightText: 2020, Ansible Project
+
+"""
+Directory and collection copying helpers.
+"""
+
+from __future__ import annotations
+
+import os
+import shutil
+import subprocess
+import tempfile
+
+from .config import ChangelogConfig, PathsConfig
+from .logger import LOGGER
+from .utils import detect_vcs
+
+
+class Copier:  # pylint: disable=too-few-public-methods
+    """
+    Allows to copy directories.
+    """
+
+    def copy(self, from_path: str, to_path: str) -> None:
+        """
+        Copy a directory ``from_path`` to a destination ``to_path``.
+
+        ``to_path`` must not exist, but its parent directory must exist.
+        """
+        LOGGER.debug("Copying complete directory from {!r} to {!r}", from_path, to_path)
+        shutil.copytree(from_path, to_path, symlinks=True)
+
+
+class GitCopier(Copier):  # pylint: disable=too-few-public-methods
+    """
+    Allows to copy directories that are part of a Git repository.
+    """
+
+    def copy(self, from_path: str, to_path: str) -> None:
+        LOGGER.debug("Identifying files not ignored by Git in {!r}", from_path)
+        os.mkdir(to_path, mode=0o700)
+        files = (
```

Comment:
Can we cleanly error when a user explicitly chooses `vcs: git` but git is not available or we're not in a git repository?

Reply:
24db74ab7cf3d1945f7e95be639042629400882f